src/common/types.ts

```diff
@@ -1,3 +1,4 @@
+import type { COMMA, CR, CRLF, DOUBLE_QUOTE, LF } from "../constants.ts";
```

Comment:
I've noticed that constants imported from src/constants.ts such as `COMMA`, `CR`, `CRLF`, `DOUBLE_QUOTE`, and `LF` are only being used as types in expressions like `typeof COMMA`. Given this usage pattern, I think it might be beneficial to also explicitly define these as types directly within `src/constants.ts`.

Here is a potential approach to consider:
```ts
export const CR = "\r";
export type CR = "\r";
// Similar declarations for COMMA, CRLF, DOUBLE_QUOTE, and LF
```

This change would allow us to standardize the naming conventions between the values and types, making our code cleaner and reducing potential confusion between type and value usage.

However, I have not fully assessed the implications of this change across our entire codebase. There might be side effects, especially where these constants are used in different contexts. Therefore, a detailed review of how these changes would affect existing implementations is necessary.

I believe the benefits of having unified naming for types and values could significantly streamline our type usage and improve code readability. Please let me know your thoughts on this proposal or if there are any concerns regarding potential side effects.